src/server/scripts/Commands/cs_cast.cpp

```diff
@@ -191,13 +202,7 @@ class cast_commandscript : public CommandScript
         if (!*args)
             return false;
 
-        Unit* target = handler->getSelectedUnit();
-        if (!target)
-        {
-            handler->SendSysMessage(LANG_SELECT_CHAR_OR_CREATURE);
-            handler->SetSentErrorMessage(true);
-            return false;
-        }
+        Unit* player = handler->GetSession()->GetPlayer();
```

Comment:
it makes sense if you want the target casts a spell to itself
.cast : player -> target
.cast back : target -> player
.cast target : target -> target's victim
.cast self : target -> target